CenterForOpenScience/ember-osf-web
# [EMB-566] Fix editable registrations, displaying component institutions

## Purpose
Fix the following issues with registration institutions.

- Upon transitioning from parent reg to component registration, the affiliated institutions list still shows
parent's affiliated institutions list instead of the component affiliated institutions.

- Edge case where the node affiliations exceed one page i.e > 10 institutions. To edit affiliated institutions we load current user institutions and compare them to registration's affiliated institutions
to tell which action (remove/add) the user can take. We were only passing in the first page. 

## Summary of Changes
- Fix `paginated-list` to reload when passed different model
- Fix edit mode by loading all nodes affiliated institutions
- Use sparse fields to only get registration's institutions `id`

## Side Effects

## Feature Flags

## QA Notes
Registration component's affiliated institutions:
- Have different affiliations on parent and component registrations.
- Go to parent registration, components list, hit component's registration
- The institution's list should reload
- Please check that component's affiliated institutions are correct.

Edge case
- Create a case where a registration has more than 10 affiliated institutions, some of which are not among the current user's affiliated institutions, can be done by using multiple admins.
- Click to edit affiliated institutions, make sure the modal has the list and actions (remove/add)

## Ticket

[EMB-566](https://openscience.atlassian.net/browse/EMB-566)

# Reviewer Checklist
